app/graphql/types/query_type.rb

```diff
@@ -19,14 +19,37 @@ def user(id:)
 
     # Dog Queries
     field :dogs, [Types::DogType], null: false,
-      description: "Return all dogs"
+      description: 'Return all dogs, or all dogs filtered by attribute' do
+        argument :activity_level_range, [Int], required: false
+        argument :age_range, [Float], required: false
+        argument :breed, [String], required: false
+        argument :weight_range, [Int], required: false
+      end
 
-    def dogs
-      Dog.all
+    def dogs(**filters)
+      if filters[:age_range]
+        raise 'Please provide an array with two integers or floating point numbers.' unless filters[:age_range].count == 2
```

Comment:
Let's try using the method provided by the gem instead of `raise`.

Reply:
https://graphql-ruby.org/errors/execution_errors
For example:`raise GraphQL::ExecutionError.new("Please provide an array with two integers or floating point numbers.", extensions: { "code" => 422 })`

Reply:
Might want to mention here which filter is messed up, too.

Reply:
Resolved all on branch